Source/WebCore/platform/graphics/MediaPlayer.cpp

```diff
@@ -208,7 +208,7 @@ class NullMediaPlayerClient : public MediaPlayerClient {
 
     const Vector<WebCore::ContentType>& mediaContentTypesRequiringHardwareSupport() const final { return nullContentTypeVector(); }
 
-    RefPtr<PlatformMediaResourceLoader> mediaPlayerCreateResourceLoader() final { return nullptr; }
+    RefPtr<PlatformMediaResourceLoader> mediaPlayerCreateResourceLoader() final { ASSERT_NOT_REACHED(); return nullptr; }
```

Comment:
Surprised we don't we need `NO_RETURN_DUE_TO_ASSERT` here.

Reply:
this macro doesn't appear to work unless your method returns `void`